File: burstLibrary/src/main/java/com/harry1453/burst/explorer/service/BurstBlockchainService.java
package com.harry1453.burst.explorer.service;

import com.harry1453.burst.explorer.entity.Account;
import com.harry1453.burst.explorer.entity.Block;
import com.harry1453.burst.explorer.entity.SearchResult;
import com.harry1453.burst.explorer.entity.Transaction;

import java.math.BigInteger;
import java.util.List;

import io.reactivex.Single;

public interface BurstBlockchainService {
    Single<List<Block>> fetchRecentBlocks();
    Single<Block> fetchBlockByHeight(BigInteger blockHeight);
    Single<Block> fetchBlockByID(BigInteger blockID);
    Single<Account> fetchAccount(BigInteger accountID);
    Single<BigInteger> fetchAccountRewardRecipient(BigInteger accountID);
    Single<List<BigInteger>> fetchAccountTransactions(BigInteger accountID);
    Single<Transaction> fetchTransaction(BigInteger transactionID);
    Single<SearchResult> determineSearchRequestType(String searchRequest);
}

File: burstLibrary/src/main/java/com/harry1453/burst/explorer/service/BurstBlockchainService.kt
package com.harry1453.burst.explorer.service

import com.harry1453.burst.explorer.entity.Account
import com.harry1453.burst.explorer.entity.Block
import com.harry1453.burst.explorer.entity.SearchResult
import com.harry1453.burst.explorer.entity.Transaction

import java.math.BigInteger

import io.reactivex.Single

interface BurstBlockchainService {
    fun fetchRecentBlocks(): Single<List<Block>>
    fun fetchBlockByHeight(blockHeight: BigInteger): Single<Block>
    fun fetchBlockByID(blockID: BigInteger): Single<Block>
    fun fetchAccount(accountID: BigInteger): Single<Account>
    fun fetchAccountRewardRecipient(accountID: BigInteger): Single<BigInteger>
    fun fetchAccountTransactions(accountID: BigInteger): Single<List<BigInteger>>
    fun fetchTransaction(transactionID: BigInteger): Single<Transaction>
    fun determineSearchRequestType(searchRequest: String): Single<SearchResult>
}
